tests/test_filters.py

```diff
@@ -527,6 +524,33 @@ def test_glyph():
         sphere.glyph(geom=geoms, indices=indices[:-1])
 
 
+def test_glyph_cell_point_data():
+
+    sphere = pyvista.Sphere(radius=3.14, theta_resolution=5, phi_resolution=5)
+    # make cool swirly pattern
+    vectors = np.vstack((np.sin(sphere.points[:, 0]),
+                        np.cos(sphere.points[:, 1]),
+                        np.cos(sphere.points[:, 2]))).T
+
+    sphere_center_points = sphere.cell_centers()
+    vectors_centers = np.vstack((np.sin(sphere_center_points.points[:, 0]),
+                        np.cos(sphere_center_points.points[:, 1]),
+                        np.cos(sphere_center_points.points[:, 2]))).T
+
+    sphere_center = sphere.copy()
+    sphere_center["vectors_cell"] = vectors_centers*0.3
+    sphere_center["vectors_points"] = vectors*0.3
+    sphere_center['arr_cell'] = np.ones(sphere.n_cells)
+    sphere_center['arr_points'] = np.ones(sphere.n_points)
+
+    
+    assert sphere_center.glyph(orient="vectors_cell", scale = "arr_cell")
+    assert sphere_center.glyph(orient="vectors_points", scale = "arr_points")
+    with pytest.raises(ValueError):
+        sphere_center.glyph(orient="vectors_cell",scale = 'arr_cell')
+        sphere_center.glyph(orient="vectors_cell",scale = "arr_points")
+        sphere_center.glyph(orient="vectors_points", scale="arr_cell")
```

Comment:
I won't be able to review in detail for a while, but this caught my eye: we should have three separate context managers, one for each call that should raise. Otherwise a single one raising would satisfy pytest even if the other two wouldn't raise. (Whitespace and quote consistency also needs some cleanup, but that's for a proper review.)

Reply:
I don't know if this has been discussed, but I find the use of black formatter to be very nice and frees up review time for more important things.

Reply:
Good catch, in fact the one line was duplicated from above and indeed doesn't raise.

Reply:
I support black in principle but it should only be introduced to the codebase after some discussion, because the corresponding changes can mess up the history (although there are partial solutions to that).